plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ServletClasspathProvider.java

```diff
@@ -100,6 +100,19 @@ public void onFailure(Throwable t) {
     return null;
   }
 
+  /** Request that a project's runtime server container be updated. */
+  private void requestClasspathContainerUpdate(
+      IProject project, IRuntime runtime, IClasspathEntry[] entries) {
+    /* Updating a container is done by either explicitly requesting an update from a
+     * {@code ClasspathContainerInitializer#requestClasspathContainerUpdate()} or by calling {@code
+     * JavaCore.setClasspathContainer()}. Both require knowledge of how the WTP Server Runtime
+     * containers are named. But {@code resolveClasspathContainerImpl()}'s implementation calls our
+     * {@code resolveClasspathContainer()} and calls {@code JavaCore.setClasspathContainer()} if the
+     * classpath entries change. */
```

Comment:
Honestly, I cannot get what the comment is trying to say, particularly "both require knowledge of how the WTP Server Runtime containers are named".

I think it's a good idea to add a link to your comment https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/3055#issuecomment-390242592 too.

Reply:
Will rework the comment and link to this issue.